language-server/src/main/scala/scala/meta/languageserver/ScalametaLanguageServer.scala

```diff
@@ -5,52 +5,87 @@ import java.io.OutputStream
 import java.io.PrintStream
 import java.nio.charset.StandardCharsets
 import java.nio.file.Files
+import scala.collection.mutable.ListBuffer
+import scala.meta.languageserver.ScalametaEnrichments._
 import scala.util.control.NonFatal
-import scalafix.languageserver.ScalafixLintProvider
 import langserver.core.LanguageServer
 import langserver.messages.ClientCapabilities
+import langserver.messages.DefinitionResult
 import langserver.messages.FileChangeType
 import langserver.messages.FileEvent
 import langserver.messages.MessageType
 import langserver.messages.ServerCapabilities
 import langserver.types._
+import monix.execution.Cancelable
+import monix.execution.Scheduler
+import monix.reactive.MulticastStrategy
+import monix.reactive.Observable
+import monix.reactive.Observer
+import monix.reactive.OverflowStrategy
+import org.langmeta.internal.semanticdb.schema
 import org.langmeta.io.AbsolutePath
+import org.langmeta.semanticdb.Database
+import org.langmeta.semanticdb.Denotation
 
 class ScalametaLanguageServer(
     cwd: AbsolutePath,
-    in: InputStream,
-    out: OutputStream,
-    scalafmt: Formatter
-) extends LanguageServer(in, out) {
-  val ps = new PrintStream(out)
-  val buffers = Buffers()
-  val scalafixService = new ScalafixLintProvider(cwd, out, connection)
-  def read(path: AbsolutePath): String =
+    lspIn: InputStream,
+    lspOut: OutputStream,
+    stdout: PrintStream
+)(implicit s: Scheduler)
+    extends LanguageServer(lspIn, lspOut) {
+  implicit val workspacePath: AbsolutePath = cwd
+  val (semanticdbSubscriber, semanticdbPublisher) =
+    ScalametaLanguageServer.semanticdbStream
+  val buffers: Buffers = Buffers()
+  val symbol: SymbolIndexer =
```

Comment:
I would call this `symbolIndexer`. `symbol` is a bit confusing, given the context